plugins/scaffolder/src/api.ts

```diff
@@ -14,377 +14,12 @@
-
-    return new ObservableImpl(subscriber => {
-      this.discoveryApi.getBaseUrl('scaffolder').then(async baseUrl => {
-        while (!subscriber.closed) {
-          const url = `${baseUrl}/v2/tasks/${encodeURIComponent(
-            taskId,
-          )}/events?${qs.stringify({ after })}`;
-          const response = await this.fetchApi.fetch(url);
-
-          if (!response.ok) {
-            // wait for one second to not run into an
-            await new Promise(resolve => setTimeout(resolve, 1000));
-            continue;
-          }
-
-          const logs = (await response.json()) as LogEvent[];
-
-          for (const event of logs) {
-            after = Number(event.id);
-
-            subscriber.next(event);
-
-            if (event.type === 'completion') {
-              subscriber.complete();
-              return;
-            }
-          }
-        }
-      });
-    });
-  }
-
-  async listActions(): Promise<ListActionsResponse> {
-    const baseUrl = await this.discoveryApi.getBaseUrl('scaffolder');
-    const response = await this.fetchApi.fetch(`${baseUrl}/v2/actions`);
-    if (!response.ok) {
-      throw await ResponseError.fromResponse(response);
-    }
-
-    return await response.json();
-  }
-
-  async cancelTask(taskId: string): Promise<void> {
-    const baseUrl = await this.discoveryApi.getBaseUrl('scaffolder');
-    const url = `${baseUrl}/v2/tasks/${encodeURIComponent(taskId)}/cancel`;
-
-    const response = await this.fetchApi.fetch(url, {
-      method: 'POST',
-    });
-
-    if (!response.ok) {
-      throw await ResponseError.fromResponse(response);
-    }
-
-    return await response.json();
-  }
-
-  async retry?(taskId: string): Promise<void> {
-    const baseUrl = await this.discoveryApi.getBaseUrl('scaffolder');
-    const url = `${baseUrl}/v2/tasks/${encodeURIComponent(taskId)}/retry`;
-
-    const response = await this.fetchApi.fetch(url, {
-      method: 'POST',
-    });
-
-    if (!response.ok) {
-      throw await ResponseError.fromResponse(response);
-    }
-
-    return await response.json();
-  }
-
-  async autocomplete({
-    token,
-    resource,
-    provider,
-    context,
-  }: {
-    token: string;
-    provider: string;
-    resource: string;
-    context?: Record<string, string>;
-  }): Promise<{ results: { title?: string; id: string }[] }> {
-    const baseUrl = await this.discoveryApi.getBaseUrl('scaffolder');
-
-    const url = `${baseUrl}/v2/autocomplete/${provider}/${resource}`;
-
-    const response = await this.fetchApi.fetch(url, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        token,
-        context: context ?? {},
-      }),
-    });
-
-    if (!response.ok) {
-      throw await ResponseError.fromResponse(response);
-    }
-
-    const { results } = await response.json();
-    return { results };
-  }
-}
+export class ScaffolderClient extends _ScaffolderClient {}
```

Comment:
We could probably just do a `export {ScaffolderClient} from '@backstage/plugin-scaffolder-common'` here instead. Example, https://github.com/backstage/backstage/blob/2d49ad89ba39fd78b0c44aa19386c41698274d2e/plugins/auth-backend/src/providers/oauth2-proxy/index.ts#L18-L24

Reply:
I had attempted to do that, but reverted because I didn't get a nice IntelliSense experience for the deprecation. So, if I do this:
```ts
export {
  /**
   * @deprecated use import from {@link @backstage/plugin-scaffolder-common#ScaffolderClient} instead as this has now been moved.
   */
  ScaffolderClient 
} from '@backstage/plugin-scaffolder-common';
```
Anywhere that imports  `ScaffolderClient` from `@backstage/plugin-scaffolder-react` will show up like this in VS Code, which is fine:
<img width="630" alt="image" src="https://github.com/user-attachments/assets/26a66431-b2d0-4a13-904c-e783fb87b52c" />
The problems are:
1. it doesn't mention the `@deprecated` comment, which it would otherwise show like this:
   <img width="803" alt="image" src="https://github.com/user-attachments/assets/e7663df6-0f26-4cb1-a20f-d17f57b60312" />

2. ⌘+clicking the `ScaffolderClient` takes me to the export from `@backstage/plugin-scaffolder-common`:
   <img width="811" alt="image" src="https://github.com/user-attachments/assets/38bbab2e-ba19-4669-8208-e6c7a0d40ea8" />
   <img width="798" alt="image" src="https://github.com/user-attachments/assets/7b9b897a-eb20-4f97-b932-b3153c02dceb" />
as opposed to taking me to the deprecated export, like so:
   <img width="780" alt="image" src="https://github.com/user-attachments/assets/866b549e-8740-4400-8021-89782422c01c" />

If I'm being too picky here, I'm happy to match the example you shared.

Reply:
Hmm, can you try https://github.com/backstage/backstage/blob/821f065ee9dc781de74c1cfa7ce38b751f1e35b2/packages/backend-common/src/deprecated/index.ts#L119-L130? That's the only example I have of re-exporting and marking as deprecated.

Reply:
Thank you for looking into it and finding that example. What this will do, though, is that it will keep the old implementation of `ScaffolderClient`. How about this example?
https://github.com/solimant/backstage/blob/104f83587b922fadbcae22396ea03398e0b63f8e/plugins/catalog-backend-module-github/src/deprecated.ts#L33-L47
It uses the same pattern of having the deprecated type extend the new type.